bin/node-esm

```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env sh
+exec node -r esm $*
```

Comment:
This does not work with global installs at the moment, because the -r esm does not resolve to the global esm install.

Not sure what the best way of fixing that is. At first thought, two major options are apparent to me:
A. I have deployed a wide wide variety of ridiculous crazy shell scripting hacks in the past to detect where the source script (`node-esm`) is located but every single one has left me feeling not-great. They could be deployed here.
B. Alternatively, this script could itself be converted to be a node script that uses `__dirname` to locate the `esm` install. This would be pretty foolproof, but I don't have a good handle on what such a loader would look like. Is it just updating process.argv (to make it look like the target is the thing that got launched, not this loader script), require("esm"), load the target & go? Maybe worth a shot but like A. I have cold-feet.

Both of these paths intimidate me, so I left this PR at a broken-but-conceptually clean & understandable state. I tend a little towards A., although I'm kind of intimidated by all the corner cases that need testing to get it working right. B. might be simple & crazy cross platform but I feel even less well versed in knowing what corner cases I'd need to take care of.

Reply:
Just an FYI: to preserve spaces and other explosion-prone things in the arguments faithfully, use `"$@"` next time.